burnman/eos/hp.py

```diff
@@ -153,7 +153,12 @@ def gibbs_free_energy(self,pressure,temperature, volume, params):
             else:
                 Gdisord=0.0
 
-        return params['H_0'] + self.__intCpdT(temperature, params) - temperature*(params['S_0'] + self.__intCpoverTdT(temperature, params)) + intVdP + Gdisord
+        if params.has_key('magnetic_moment'):
+            Gmagnetic=self._magnetic_gibbs(pressure, temperature, params)
+        else:
+            Gmagnetic=0.0
+
+        return params['H_0'] + self.__intCpdT(temperature, params) - temperature*(params['S_0'] + self.__intCpoverTdT(temperature, params)) + intVdP + Gdisord + Gmagnetic
```

Comment:
You give me too much philosophical credit.  :)

I don't mean necessarily setting it to zero, I just mean that it would be more consistent to do this kind of check with the others.  I don't really have a specific suggestion for how to handle it, so don't worry about it too much.  

You are right that we don't want to do unnecessary computations.